src/core/stream_recv.c

```diff
@@ -162,6 +162,39 @@ QuicStreamRecvQueueFlush(
     }
 }
 
+//
+// Processes a received RELIABLE_RESET frame's payload.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicStreamProcessReliableResetFrame(
+    _In_ QUIC_STREAM* Stream,
+    _In_ uint64_t FinalSize,
+    _In_ QUIC_VAR_INT ErrorCode,
+    _In_ QUIC_VAR_INT ReliableOffset
+    )
+{
+    if (Stream->ReliableOffsetRecv == 0 || ReliableOffset < Stream->ReliableOffsetRecv) {
+        //
+        // As outlined in the spec, if we receive multiple CLOSE_STREAM frames, we only accept strictly
+        // decreasing offsets.
+        //
+        Stream->ReliableOffsetRecv = ReliableOffset; // TODO: Remove ReliableOffsetRecv once everything works.
+        Stream->RecvMaxLength = ReliableOffset;
+    }
+
+
+    if (Stream->RecvBuffer.BaseOffset >= Stream->RecvMaxLength) {
+        //
+        // All data delivered. Queue flush.
+        //
+        Stream->Flags.ReceiveDataPending = TRUE;
+        QuicStreamRecvQueueFlush(
```

Comment:
You don't need to queue a recv flush. Instead just probably deliver the shutdown callback.

Reply:
Added, and I also added an event indication.